BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java

```diff
@@ -131,28 +132,40 @@ public void receiveDetections(Detector.Detections<Barcode> detections) {
   }
 
   private void handleIsbnInput(String isbn) {
-    isbnRetriever = new IsbnRetriever(isbn);
-    System.out.println(isbnRetriever);
-    thread = new Thread(isbnRetriever);
-    thread.start();
-
-    try {
-      thread.join();
-    } catch (Exception e) {
-      System.out.println(e);
-    }
+    String cleanIsbn = isbn.replaceAll("\\s", "");
+
+    if (dataValidation.isValidIsbn10or13(cleanIsbn)) {
```

Comment:
```suggestion
    if (DataValidation.isValidIsbn10or13(cleanIsbn)) {
```
Access the static method this way instead through an instance.

Reply:
Sure, didn't think of this - thanks!